presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PartialAggregationPushDown.java

```diff
@@ -90,7 +90,7 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Void> cont
                 return context.defaultRewrite(node);
             }
 
-            boolean decomposable = node.isDecomposable(functionRegistry);
+            boolean decomposable = node.isDecomposable(functionRegistry) && node.getOrderBySymbols().isEmpty();
```

Comment:
This will brake `AddExchanges` for distributed ROLLUP/CUBE. After your change, your code will report to `AddExchanges` that `AggregationNode` has default output and IS decomposable, `AddExchanges` will distributed aggregation assuming that there will be `PARTIAL` aggregation, but this line of code, will prevent this optimizer to create this `PARTIAL` aggregation that is required for correctness.

Probably you will be fine by placing ` && node.getOrderBySymbols().isEmpty()` inside `AggregationNode.isDecomposable(...)` method.

However, please add tests for ROLLUP/CUBE with ORDER BY in `AbstractTestQueries`